### DISCUSSION TITLE
Why is it necessary to have both public and private buckets?

### DISCUSSION THREAD

**Discussion by toabi** [theriverman/django-minio-backend#20] [Q&A]
If an application only uses one type of bucket it seems weird to have to define both.

**Comment by theriverman** [comment]
thanks for the question!

I've implemented this safety guard with the intention of reducing developer/maintainer debug times.
if you believe that this approach does not fulfil its purpose, then I'm very much open to further discussion!

**Reply by MRigal** [reply]
Same for me here! I use MinIO only to save some kind of data inside an intranet and I don't want public buckets for it. I have to use a subclass of the Backend to overcome this safety guard, that's pretty weird!

**Reply by toabi** [reply]
We also usually only need public *OR* private files in our projects. In all those project it's highly annoying that this code forces the existence of two buckets.

**Comment by theriverman** [comment]
I'm closing this discussion as it has been solved in the code base.

**Reply by MRigal** [reply]
Thanks, a pity that issues and discussions are not well kept in sync in Github.... The discussion was followed by https://github.com/theriverman/django-minio-backend/issues/29 which was fixed in 3.2.0 for other followers here...
